client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -170,8 +170,13 @@ public void channelUnregistered(ChannelHandlerContext ctx) throws Exception {
         }
         channel.set(null);
         flowIdReplyProcessorMap.forEach((flowId, rp) -> {
-            rp.connectionDropped();
-            log.debug("Connection dropped for flow id {}", flowId);
+            try {
+                log.debug("Connection dropped for flow id {}", flowId);
+                rp.connectionDropped();
+            } catch (Exception e) {
+                // Suppressing exception which prevents all ReplyProcessor.connectionDropped from being invoked.
+                log.warn("Encountered exception invoking ReplyProcessor for flow id " + flowId, e);
```

Comment:
Is any exception caught here equally important? Just thinking in the case that there could be expected and harmless exceptions being caught here, so we could change the log level and do no print the whole stack trace for them (for the rest, this way of logging may be ok).

Reply:
Ideally exceptions should not be thrown by the ReplyProcessor implementations (the current ReplyProcessors do not throw it too). If thrown then it should be logged as it indicates that the newer ReplyProcessor s are incorrect and we prevent breakages with the future version of ReplyProcessors. With this in context feel that WARN with the stack trace is helpful.